glusterd2/commands/volumes/grouped-options.go

```diff
@@ -5,50 +5,132 @@ import (
 )
 
 // defaultGroupOptions maps from a profile name to a set of options
-var defaultGroupOptions = map[string][]api.VolumeOption{
-	"profile.gluster-block": {{"performance.quick-read", "off", "on"},
-		{"performance.read-ahead", "off", "on"},
-		{"performance.io-cache", "off", "on"},
-		{"performance.stat-prefetch", "off", "on"},
-		{"performance.open-behind", "off", "on"},
-		{"performance.readdir-ahead", "off", "on"},
-		{"performance.strict-o-direct", "on", "off"},
-		{"network.remote-dio", "disable", "enable"},
-		{"cluster.eager-lock", "disable", "enable"},
-		{"cluster.quorum-type", "auto", "none"},
-		{"cluster.data-self-heal-algorithm", "full", "reset"},
-		{"cluster.locking-scheme", "granular", "full"},
-		{"cluster.shd-max-threads", "8", "1"},
-		{"cluster.shd-wait-qlength", "10000", "1024"},
-		{"features.shard", "on", "off"},
-		{"features.shard-block-size", "64MB", "512MB"},
-		{"user.cifs", "off", "on"},
-		{"server.allow-insecure", "on", "off"}},
-	"profile.nl-cache": {{"features.cache-invalidation", "on", "off"},
-		{"features.cache-invalidation-timeout", "600", "60"},
-		{"performance.nl-cache", "on", "off"},
-		{"performance.nl-cache-timeout", "600", "600"},
-		{"network.inode-lru-limit", "50000", "16384"}},
-	"profile.metadata-cache": {{"features.cache-invalidation", "on", "off"},
-		{"features.cache-invalidation-timeout", "600", "60"},
-		{"performance.stat-prefetch", "on", "off"},
-		{"performance.cache-invalidation", "on", "off"},
-		{"performance.md-cache-timeout", "600", "600"},
-		{"network.inode-lru-limit", "50000", "16384"}},
-	"profile.virt": {{"performance.quick-read", "off", "on"},
-		{"performance.read-ahead", "off", "on"},
-		{"performance.io-cache", "off", "on"},
-		{"performance.low-prio-threads", "32", "32"},
-		{"network.remote-dio", "enable", "disable"},
-		{"cluster.eager-lock", "enable", "disable"},
-		{"cluster.quorum-type", "auto", "none"},
-		{"cluster.server-quorum-type", "server", "none"},
-		{"cluster.data-self-heal-algorithm", "full", "reset"},
-		{"cluster.locking-scheme", "granular", "full"},
-		{"cluster.shd-max-threads", "8", "1"},
-		{"cluster.shd-wait-qlength", "10000", "1024"},
-		{"features.shard", "on", "off"},
-		{"user.cifs", "off", "on"}},
-	"profile.test": {{"afr.eager-lock", "on", "off"},
-		{"gfproxy.afr.eager-lock", "on", "off"}},
+var defaultGroupOptions = map[string]*api.OptionGroup{
```

Comment:
Can this be modified to read all groups from external json/yml file from predefined path(`<localstatedir>/group_profiles`)? 

    func GetDefaultGroupOptions()

Reply:
ok, will do that as a dependent patch